modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -291,39 +300,104 @@
-                                        (some-> ^TimePartitioning (tabledef->time-partition tabledef) .getField)))
-        fields                    (set
-                                   (map
-                                    #(assoc % :database-partitioned (= (:name %) partitioned-field-name))
-                                    (fields->metabase-field-info (.. tabledef getSchema getFields))))]
-    {:schema dataset-id
-     :name   table-name
-     :fields (cond-> fields
-               ;; if table has time partition but no field is specified as partitioned
-               ;; meaning this table is partitioned by ingestion time
-               ;; so we manually sync the 2 pseudo-columns _PARTITIONTIME AND _PARTITIONDATE
-               (and is-partitioned?
-                    (some? (tabledef->time-partition tabledef))
-                    (nil? partitioned-field-name))
-               (conj
-                {:name                 partitioned-time-field-name
-                 :database-type        "TIMESTAMP"
-                 :base-type            (bigquery-type->base-type nil "TIMESTAMP")
-                 :database-position    (count fields)
-                 :database-partitioned true}
-                {:name                 partitioned-date-field-name
-                 :database-type        "DATE"
-                 :base-type            (bigquery-type->base-type nil "DATE")
-                 :database-position    (inc (count fields))
-                 :database-partitioned true}))}))
+(defn- build-nested-column-lookup
+  "Returns a map of table-name->parent-path->nested-columns"
+  [driver database project-id dataset-id table-names]
+  (let [results (query-honeysql
+                 driver
+                 database
+                 (cond->
+                  {:select [:table_name :column_name :data_type :field_path]
+                   :from [[(information-schema-table project-id dataset-id "COLUMN_FIELD_PATHS") :c]]}
+                   (not-empty table-names)
+                   (assoc :where [:in :table_name table-names])))
+        nested-columns (map (fn [{data-type :data_type field-path-str :field_path table-name :table_name}]
+                              (let [field-path (str/split field-path-str #"\.")
+                                    nfc-path (not-empty (pop field-path))
+                                    [database-type base-type] (raw-type->database+base-type data-type)]
+                                {:name (peek field-path)
+                                 :table-name table-name
+                                 :table-schema dataset-id
+                                 :database-type database-type
+                                 :base-type base-type
+                                 :nfc-path nfc-path}))
+                            results)]
+    (reduce
+     (fn [accum col]
+       (when-let [parent (:nfc-path col)]
+         (update-in accum [(:table-name col) parent] (fnil conj []) col)))
```

Comment:
I expected a `let` and a `cond-> accum parent ...` combination. Is there a test covering this?

Reply:
Good catch I don't know how `sync-nested-fields-test` missed it.